src/SparkFun_u-blox_GNSS_Arduino_Library.cpp

```diff
@@ -2781,6 +2852,56 @@ void SFE_UBLOX_GNSS::sendSerialCommand(ubxPacket *outgoingUBX)
   _serialPort->write(outgoingUBX->checksumB);
 }
 
+
+// Transfer a byte to SPI. Also capture any bytes received from the UBLOX device during sending and capture them in a small buffer so that
+// they can be processed later with process
+void SFE_UBLOX_GNSS::spiTransfer(uint8_t byteToTransfer) 
+{
+  uint8_t returnedByte = _spiPort->transfer(byteToTransfer);
+  if (returnedByte != 0xFF)
+  {
+    spiBuffer[spiBufferIndex] = returnedByte;
+    spiBufferIndex++;
```

Comment:
Question: where does ```spiBufferIndex``` get reset to zero?

Reply:
It gets cleared out in checkUbloxSPI. Comments below...